presto-native-execution/src/test/java/com/facebook/presto/nativeworker/TestHiveQueriesGlueMetastore.java

```diff
@@ -0,0 +1,446 @@
+import com.facebook.presto.testing.QueryRunner;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.io.Files;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Paths;
+import java.util.Map;
+import java.util.Optional;
+import java.util.UUID;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.google.common.io.RecursiveDeleteOption.ALLOW_INSECURE;
+import static java.util.Locale.ENGLISH;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertNotNull;
+
+/**
+ * TestHiveGlueMetastoreUsingThrift requires AWS credentials to be correctly set up in order to access Glue catalog.
+ * You can configure credentials locally in ~/.aws/credentials before running the test. Alternatively, you can set up
+ * AWS access/secret keys or IAM role in `setUpGlueCatalog` method in the test suite using the Glue config properties
+ * from https://prestodb.io/docs/current/connector/hive.html#aws-glue-catalog-configuration-properties.
+ */
+abstract class TestHiveQueriesGlueMetastore
+        extends TestHiveQueries
+{
+    private static final String TEMPORARY_PREFIX = "tmp_presto_glue_test_";
+    // Catalog name used in the tests, it is also hardcoded in HiveExternalWorkerQueryRunner so workers can register
+    // the connector so if you want to change the name, also update it in HiveExternalWorkerQueryRunner.
+    private static final String CATALOG_NAME = "test_glue";
+
+    // Warehouse directory for the Glue catalog.
+    private File tempWarehouseDir;
+    // Fully-qualified schema name in format [catalog].[schema] used in tests.
+    private String schemaName;
+
+    private final boolean useThrift;
+
+    protected TestHiveQueriesGlueMetastore(boolean useThrift)
+    {
+        super(useThrift);
+        this.useThrift = useThrift;
+    }
+    @Override
+    protected QueryRunner createQueryRunner() throws Exception
+    {
+        String prestoServerPath = System.getProperty("PRESTO_SERVER");
+        String dataDirectory = System.getProperty("DATA_DIR");
+        String workerCount = System.getProperty("WORKER_COUNT");
+        int cacheMaxSize = 0;
+
+        assertNotNull(prestoServerPath, "Native worker binary path is missing. Add -DPRESTO_SERVER=<path/to/presto_server> to your JVM arguments.");
+        assertNotNull(dataDirectory, "Data directory path is missing. Add -DDATA_DIR=<path/to/data> to your JVM arguments.");
+
+        return HiveExternalWorkerQueryRunner.createNativeQueryRunner(
+                dataDirectory,
+                prestoServerPath,
+                Optional.ofNullable(workerCount).map(Integer::parseInt),
+                cacheMaxSize,
+                useThrift);
+    }
+
+    @Override
+    protected ExpectedQueryRunner createExpectedQueryRunner() throws Exception
```

Comment:
Again, why define this function again? It's the same as the the one in AbstractTestHiveQueries which is overriden.

Reply:
Updated. This was done to enhance the query runner with the catalog setup and session configuration which was removed.